### ISSUE TITLE
[Bug] Crash on boot

### ISSUE THREAD

**Issue by davic-anime** [Fleker/CumulusTV#254]
Hi Fleker,
There is a crash if I start Live Channels after a full reboot of the Android TV box (Mi Box). Only way to bypass that crash is by first booting Cumulus TV before starting Live Channels after a reboot.

**Comment by JeanPrendnote** [comment]
yea i have the same problem

**Comment by Fleker** [comment]
I will check crash reports for what's happening. There's probably some Null
pointer exception.

**Comment by Fleker** [comment]
I'm going to initialize some data when the device first boots to prevent this from causing crashes. I think that's what's happening.

**Comment by loueradun** [comment]
If this is the same issue as #260, then it seems to have been fixed in 1.7.8.  Not getting any crashes on reboot or resuming from sleep anymore.

**Comment by Fleker** [comment]
It's good to hear the bug has been resolved. I haven't heard any other comments. I will be closing this.

**Comment by loueradun** [comment]
@Fleker It looks like this issue may not be 100% resolved in all cases... It had been running good, and then I saw this issue again just the other day.  If I happen to find out exactly what situation brings about the issue I'll report back.

**Comment by Fleker** [comment]
😖

**Comment by loueradun** [comment]
I'm not entirely sure what is going on here.  I can reboot my Shield, launch Cumulus and everything works.  I can put my Shield to sleep and resume it, and everything works.  If I put the Shield to sleep and it sits for a while, that's when I seem to start having issues.  Most of the time I get the "Your channel will be available shortly" message, but other times it will crash in an endless loop (requires force close of live channels to stop endless launching of Cumulus).  That said, it is much better than it used to be.  Any reboot/sleep would cause the issue previously.  I think now it may be due to some garbage collection after the device has sat idle for a while, and proper detection of that state.

**Comment by Fleker** [comment]
Yes it's definitely saving things in memory and it is having issues recovering that save. It's hard to find a solution since it's hard to reliably reproduce.

**Comment by Fleker** [comment]
I think I can cache a lot more stuff and mark certain channels as temporary to be flushed out when the system resyncs. That way there won't be any nulls.
